src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4821,6 +4824,20 @@ jint os::init_2(void) {
     }
   }
 
+  // Check the availability of MADV_POPULATE_WRITE.
+  if (UseMadvPopulateWrite) {
+    // Some downstream kernels recognize MADV_POPULATE_WRITE_value as another
+    // advice, so the check of versions is required here.
+    // See https://github.com/oracle/linux-uek/issues/23
+    bool supportMadvPopulateWrite =
+      (Linux::_release_major > 5 ||
+       (Linux::_release_major == 5 && Linux::_release_minor >= 14)) &&
+      (::madvise(0, 0, MADV_POPULATE_WRITE) == 0);
+    if (!FLAG_IS_DEFAULT(UseMadvPopulateWrite) && !supportMadvPopulateWrite)
+      warning("Platform is supposed not to support MADV_POPULATE_WRITE, "
+              "disabling using it to pretouch (-XX:-UseMadvPopulateWrite)");
```

Comment:
I don't see the code that disables UseMadvPopulateWrite. Is it missing?

The warning message is a bit awkward. Could it be changed to something like: "Platform does not support MADV_POPULATE_WRITE, disabling using it to pretouch memory (-XX:-UseMadvPopulateWrite)"

Could you also make sure to use `{}` even for single statement if blocks?